src/Models/BaseElement.php

```diff
@@ -694,14 +699,48 @@ public function getSummary()
         return '';
     }
 
+
+    /**
+     * @return array
+     */
+    public function getSummaryData()
+    {
+        $summaryData = [];
+
+        $this->extend('setDefaultSummaryData', $summaryData);
```

Comment:
The usual convention would be `$this->extend('updateSummaryData', $summaryData);`